cdap-proto/src/main/java/co/cask/cdap/proto/codec/DefaultThrowableCodec.java

```diff
@@ -35,18 +38,23 @@ public JsonElement serialize(DefaultThrowable src, Type typeOfSrc, JsonSerializa
     json.addProperty("className", src.getClassName());
     json.addProperty("message", src.getMessage());
     json.add("stackTraces", context.serialize(src.getStackTraces(), StackTraceElement[].class));
-
-    DefaultThrowable cause = src.getCause();
-    if (cause != null) {
-      json.add("cause", context.serialize(cause, DefaultThrowable.class));
-    }
-
+    json.add("cause", context.serialize(src.getCause(), DefaultThrowable.class));
     return json;
   }
 
   @Override
   public DefaultThrowable deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
     throws JsonParseException {
-    return context.deserialize(json, DefaultThrowable.class);
+    JsonObject jsonObj = json.getAsJsonObject();
+    String className = jsonObj.get("className").getAsString();
+    String message = jsonObj.get("message").getAsString();
+    JsonArray stackTraces = jsonObj.get("stackTraces").getAsJsonArray();
+    StackTraceElement[] stackTraceElements = context.deserialize(stackTraces, StackTraceElement[].class);
+    final JsonElement cause = jsonObj.get("cause");
+    if (cause == null) {
```

Comment:
this can be simplified a little:
```
DefaultThrowable failureCause = cause == null ? null : context.deserialize(cause, DefaultThrowable.class);
return new DefaultThrowable(className, message, stackTraceElements, failureCause);
```